src/apps/code-editor/src/app/components/RecentFiles/index.tsx

```diff
@@ -0,0 +1,106 @@
+import { useEffect, useState, useRef } from "react";
+import { Box } from "@mui/material";
+
+import { TopBar } from "./TopBar";
+import { FileList, FileProps } from "./FileList";
+import { DevResources } from "./DevResources";
+import { useSelector } from "react-redux";
+import { AppState } from "../../../../../../shell/store/types";
+import { NavCodeTypes } from "../constants";
+
+type RecentFilesProps = {
+  openCreateFileDialog: (
+    type: string,
+    nav: "view" | "stylesheet" | "script" | "file"
+  ) => void;
+};
+
+export const RecentFiles = ({ openCreateFileDialog }: RecentFilesProps) => {
+  const searchInputRef = useRef(undefined);
+  const files = useSelector((state: AppState) => state?.navCode?.raw);
+  const [recentFiles, setRecentFiles] = useState([]);
+  const [searchKeyword, setSearchKeyword] = useState("");
+
+  useEffect(() => {
+    if (!files?.length) return;
+
+    const sortedFiles: FileProps[] = files
+      ?.sort(
```

Comment:
Unnecessary optional chaining, you're already checking if files has a value above.